deploy/template/deployment.yaml

```diff
@@ -70,6 +78,57 @@ metadata:
   name: system:cloud-controller-manager
 rules:
 - apiGroups:
+  # reason: so ccm can create metallb-system namespace
+  - ""
+  resources:
+  - namespaces
+  verbs:
+  - create
+  - patch
```

Comment:
If we're only creating namespaces then I get the `list / get` to determine if they exist along with the `create` but do we need `patch`?

Reply:
Probably not. The [code itself](https://github.com/packethost/packet-ccm/blob/master/util/manifests.go#L41-L48) is a similar loop for everything, even namespaces:

```go
		case *v1.Namespace:
			intf := client.CoreV1().Namespaces()
			existing, err := intf.Get(o.Name, metav1.GetOptions{})
			if err == nil && existing != nil {
				_, err2 = intf.Update(o)
			} else {
				_, err2 = intf.Create(o)
			}
```

If we removed the `intf.Update()` part, basically saying, "well, it exists, so I am happy", then we could get rid of the `patch` and `update`, at the cost of the code being slightly different for this `case` over every other.

I am ok either way - reduced perms with slightly more complex code vs simpler code with higher perms. What do you think @thebsdbox ?

Reply:
Ah if we're `updating` then I'm happy the way things are. :-)